appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/cdi/FaultToleranceExtension.java

```diff
@@ -0,0 +1,131 @@
+import org.eclipse.microprofile.faulttolerance.CircuitBreaker;
+import org.eclipse.microprofile.faulttolerance.Fallback;
+import org.eclipse.microprofile.faulttolerance.Retry;
+import org.eclipse.microprofile.faulttolerance.Timeout;
+
+/**
+ * CDI Extension that does the setup for FT interceptor handling.
+ * 
+ * @author Andrew Pielage
+ * @author Jan Bernitt (2.0)
+ */
+public class FaultToleranceExtension implements Extension {
+
+    /**
+     * The {@link FaultTolerance} "instance" we use to dynamically mark methods at runtime that should be
+     * handled by the {@link FaultToleranceInterceptor} that handles all of the FT annotations.
+     */
+    private static final Annotation MARKER = () -> FaultTolerance.class;
+
+    private static final String INTERCEPTOR_PRIORITY_PROPERTY = "mp.fault.tolerance.interceptor.priority";
+
+    void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscovery, BeanManager beanManager) {
+        beforeBeanDiscovery.addAnnotatedType(beanManager.createAnnotatedType(FaultToleranceInterceptor.class), "MP-FT");
+    }
+
+    /**
+     * Marks all {@link Method}s *affected* by FT annotation with the {@link FaultTolerance} annotation which
+     * is handled by the {@link FaultToleranceInterceptor} which processes the FT annotations.
+     * 
+     * @param processAnnotatedType type currently processed
+     */
+    <T> void processAnnotatedType(@Observes @WithAnnotations({ Asynchronous.class, Bulkhead.class, CircuitBreaker.class,
+        Fallback.class, Retry.class, Timeout.class }) ProcessAnnotatedType<T> processAnnotatedType) throws Exception {
+        boolean markAllMethods = FaultToleranceUtils
+                .isAnnotatedWithFaultToleranceAnnotations(processAnnotatedType.getAnnotatedType());
+        Class<?> targetClass = processAnnotatedType.getAnnotatedType().getJavaClass();
+        for (AnnotatedMethodConfigurator<?> methodConfigurator : processAnnotatedType.configureAnnotatedType().methods()) {
+            if (markAllMethods || FaultToleranceUtils
+                    .isAnnotatedWithFaultToleranceAnnotations(methodConfigurator.getAnnotated())) {
+                FaultTolerancePolicy.asAnnotated(targetClass, methodConfigurator.getAnnotated().getJavaMember());
+                methodConfigurator.add(MARKER);
+            }
+        }
+    }
+
+    void changeInterceptorPriority(@Observes ProcessAnnotatedType<FaultToleranceInterceptor> interceptorType) {
+        Optional<Integer> priorityOverride = ConfigProvider.getConfig().getOptionalValue(INTERCEPTOR_PRIORITY_PROPERTY,
+                Integer.class);
+        if (priorityOverride.isPresent()) {
+            interceptorType.configureAnnotatedType()
+                    .remove(annotation -> annotation instanceof Priority)
+                    .add(new PriorityLiteral(priorityOverride.get()));
+        }
+    }
+
+    public static final class PriorityLiteral extends AnnotationLiteral<Priority> implements Priority {
```

Comment:
Note to self and others: should propose literals to be in the FT spec for each annotation